doc/development/github_guide.md

```diff
@@ -122,6 +122,33 @@ a PR.
 Alternatively, you can explore GitHub CLI tool (_gh_) which allows you
 to do `git push` and create a PR in one step with `gh pr create -fw`.
 
+## Guidelines for writing a meaningful pull request
+
+A well-written pull request (PR) is critical because it improves collaboration
+and efficiency within a project. It allows reviewers to quickly understand the
+purpose and impact of the changes, making it easier to assess the quality,
+compatibility, and necessity of the code. Clear PRs reduce the risk of
+misunderstandings, help track progress, and ensure that changes are aligned with
+the project's goals. They also facilitate faster merges and better documentation,
+resulting in more stable and maintainable software.
```

Comment:
This sounds like chatGPT, I would shorten it to one sentence and put it  under `Creating a pull request`

Reply:
Shortened.